url2f44efcfa6_wangyudong_sohutw_java_sdk_2_0_tgz-pJ8-com_sohu_t_api_sdk_json_TestJ8/src/com/sohu/t/api/sdk/data/Status.java

```diff
@@ -0,0 +1,269 @@
+package com.sohu.t.api.sdk.data;
+
+import com.sohu.t.api.sdk.TBlogException;
+import com.sohu.t.api.sdk.TBlogResponse;
+import com.sohu.t.api.sdk.http.Response;
+import com.sohu.t.api.sdk.json.JSONArray;
+import com.sohu.t.api.sdk.json.JSONException;
+import com.sohu.t.api.sdk.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * User: chenxiaojian
+ * Date: 13-1-8
+ * Time: 下午12:31
+ */
+public class Status extends TBlogResponse {
+
+    public Integer transmit = null;
+    public Integer comment = null;
+    private String created_at = "";
+    private String id = "";
+    private String text = "";
+    private String source = "";
+    private boolean favorited = false;
+    private String truncated = "";
+    private String in_reply_to_status_id = "";
+    private String in_reply_to_user_id = "";
+    private String in_reply_to_screen_name = "";
+    private String in_reply_to_status_text = "";
+    private String small_pic = "";
+    private String middle_pic = "";
+    private String original_pic = "";
+    private User user = null;
+    private RetweetedStatus retweeted_status;
+    private String coordinates = null;
+    private Boolean in_reply_to_has_image = null;
+    private List<PicGroupVO> pics = null;
+    private Integer picsCount = null;
+
+    public static List<Status> constructStatuses(Response res) throws TBlogException {
+
+        return constructStatuses(res.asJSONArray());
+    }
+
+
+    public static List<Status> constructStatuses(JSONArray jsonArray) throws TBlogException {
+
+        try {
+            int size = jsonArray.length();
+            List<Status> statuses = new ArrayList<Status>(size);
+            for (int i = 0; i < size; i++) {
+                statuses.add(new Status(jsonArray.getJSONObject(i)));
+            }
+            return statuses;
+        } catch (JSONException jsone) {
+            throw new TBlogException(jsone);
+        } catch (TBlogException te) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)